frontend/src/Components/ProtectedRoute.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from 'react';
+import { Navigate } from 'react-router-dom';
+import { useLoading } from '../Loading/LoadingContext';
+import PropTypes from 'prop-types';
+import { API_BASE_URL } from '../config';
+
+
+const ProtectedRoute = ({ children }) => {
+const [isAuthenticated, setIsAuthenticated] = useState(false);
+const { isLoading, setIsLoading } = useLoading();
+const [error, setError] = useState(null);
+
+
+useEffect(() => {
+    const checkAuth = async () => {
```

Comment:
I'd suggest to execute this logic only once, I don't think we need to validate session every single time the switch route (only once per page refresh should be fine).

Feel free to address in follow up commit.